### PR TITLE
BFB-121: Gun Range Scene

### PR DESCRIPTION
**Pull request by m4rtint**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-121

**Description**
- Created a weapon range

**Screenshots / GIFs**
Capture

### CODE REVIEW

File: Assets/_BForBoss/_Core/Scripts/Managers/GunRangeWorldManager.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        [Title("Component")] 
+        [SerializeField] private FirstPersonPlayer _player = null;
+        [SerializeField] private PauseMenu _pauseMenu = null;
+        private FreezeActionsUtility _freezeActionsUtility = null;
+        private readonly StateManager _stateManager = StateManager.Instance;
+
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+        [Title("Debug")]
+        [SerializeField] private GameObject _debugCanvas;
```
Comment (unoctium1) [inline Assets/_BForBoss/_Core/Scripts/Managers/GunRangeWorldManager.cs:18]: I'm getting a null ref on this

**Review by unoctium1 — CHANGES_REQUESTED** [review]
Scene looks great, just the one null ref though. I'm also getting warnings from the dash vfx but I'm not sure that matters

**Comment by m4rtint** [comment]
> Scene looks great, just the one null ref though. I'm also getting warnings from the dash vfx but I'm not sure that matters

mmm yeah it's just letting the dev know, hey, visuals are missing, but it's not game breaking. This is the best way to let devs know imo?

File: Assets/_BForBoss/_Core/Scripts/Managers/GunRangeWorldManager.cs (hunk 1 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Core/Scripts/Managers/GunRangeWorldManager.cs:18]: fixed

**Review by unoctium1 — APPROVED** [review]
lgtm

**Pull request merged**